src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -3065,40 +3338,220 @@ namespace detail
                     }
                 }
 
+                // handle volume dependent fields
+                // if the field is volume dependent and we have not already calculated the volumes
+                if (vol_dep && !volumes_info.has_child("poly"))
```

Comment:
one more suggestion (sorry !) 

you don't need to pass in volumes_info, you can use a bool to check if you already created it during the iteration + if it doesn't exist, you can create it. If it does you can pull out the existing volume field ref. 



If it is already there, you can pull the volumes out